src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -484,4 +489,76 @@ else if (tokenType == TokenTypes.LITERAL_PRIVATE) {
         return illegalClassNames;
     }
 
+    /**
+     * Strip initial newline and preceding whitespace on each line from text block content.
+     * In order to be consistent with how javac handles this task, we have modeled this
+     * implementation after the code from:
+     * github.com/openjdk/jdk14u/blob/master/src/java.base/share/classes/java/lang/String.java
+     *
+     * @param textBlockContent the actual content of the text block.
+     * @return string consistent with javac representation.
+     */
+    public static String stripIndentAndInitialNewLineFromTextBlock(String textBlockContent) {
+        final String contentWithInitialNewLineRemoved =
+            ALL_NEW_LINES.matcher(textBlockContent).replaceFirst("");
+        final List<String> lines =
+            Arrays.asList(ALL_NEW_LINES.split(contentWithInitialNewLineRemoved));
+        final int indent = getSmallestIndent(lines);
+        final String suffix = "";
+
+        return lines.stream()
+                .map(line -> stripIndentFromLine(line, indent))
+                .collect(Collectors.joining(System.lineSeparator(), "", suffix));
+    }
+
+    /**
+     * Helper method for stripIndentAndInitialNewLineFromTextBlock, strips correct indent
+     * from string, or returns empty string if no text.
+     *
+     * @param line the string to strip indent from
+     * @param indent the amount of indent to remove
+     * @return modified string with removed indent, or empty string.
+     */
+    private static String stripIndentFromLine(String line, int indent) {
+        final int firstNonWhitespace = indexOfNonWhitespace(line);
+        final int lastNonWhitespace = line.length();
+        final int incidentalWhitespace = Math.min(indent, firstNonWhitespace);
+
+        return line.substring(incidentalWhitespace, lastNonWhitespace);
+    }
+
+    /**
+     * Helper method for stripIndentAndInitialNewLineFromTextBlock, to determine the smallest
+     * indent in a text block string literal.
+     *
+     * @param lines list of actual text block content, split by line.
+     * @return number of spaces representing the smallest indent in this text block.
+     */
+    private static int getSmallestIndent(List<String> lines) {
+        int indent = Integer.MAX_VALUE;
+        for (String line : lines) {
+            final int leadingWhitespace = indexOfNonWhitespace(line);
+            indent = Integer.min(indent, leadingWhitespace);
+        }
+        return indent;
+    }
+
+    /**
+     * Helper method to find the index of the first non-whitespace character in a string.
+     *
+     * @param line the string to find the first index of a non-whitespace character for.
+     * @return the index of the first non-whitespace character.
```

Comment:
return the index of the first non-whitespace character, or line.length if no such character found?
I don't like this behavior. Let's return -1 if it fails to find any non-whitespace character.
In this case, the method `getSmallestIndent` can be
```java
        return lines.stream()
                .mapToInt(CheckUtil::indexOfNonWhitespace)
                .filter(value -> value >= 0)
                .min()
                .orElse(0);
```

Reply:
Pitest shows that `                .filter(value -> value >= 0)` has 6 surviving mutations; I removed that line and all tests still pass, as well as pitest.

Reply:
Test cases added, but we still don't need that line.